src/Cutscenes.ts

```diff
@@ -1637,6 +1637,146 @@ export class Cutscenes<TEightBittr extends FullScreenPokemon> extends EightBittr
     }
 
     /**
+     * Cutscene for a Quick Attack attack in battle.
+     * 
+     * @param settings   Settings used for the cutscene.
+     * @param args   Settings for the routine.
+     */
+    public cutsceneBattleAttackQuickAttack(settings: IBattleCutsceneSettings, args: IBattleAttackRoutineSettings): void {
+        const attackerName: string = args.attackerName;
+        const defenderName: string = args.defenderName;
+        const attacker: IThing = this.EightBitter.BattleMover.getThing(attackerName) as IThing;
+        const defender: IThing = this.EightBitter.BattleMover.getThing(defenderName) as IThing;
+        const direction: number = attackerName === "player" ? 1 : -1;
+        const menu: IMenu = this.EightBitter.MenuGrapher.getMenu("BattleDisplayInitial") as IMenu;
+
+        let xvel: number = -7 * direction;
+        let dt: number = 20;
```

Comment:
It looks like all the events you're doing are delayed by 20. It might be easier to just have a function that runs after 20 ticks.

```typescript
this.EightBitter.TimeHandler.addEvent(
    (): void => {
        // all the delayed logic
    },
    20);
```